homeassistant/components/ecowitt/sensor.py

```diff
@@ -0,0 +1,511 @@
+        """Create callback for new sensors discovered."""
+        _LOGGER.debug("_new_sensor called")
+        data = hass.data[DOMAIN][entry.entry_id]
+        sensors = data.client.list_sensor_keys()
+        async_dispatcher_send(
+            hass, SIGNAL_ADD_ENTITIES.format(SENSOR_DOMAIN, entry.entry_id), sensors
+        )
+
+    data = hass.data[DOMAIN][entry.entry_id]
+    sensors = data.client.list_sensor_keys()
+    async_dispatcher_connect(
+        hass,
+        SIGNAL_ADD_ENTITIES.format(SENSOR_DOMAIN, entry.entry_id),
+        _async_add_ecowitt_entities,
+    )
+    async_dispatcher_connect(
+        hass, SIGNAL_NEW_SENSOR.format(SENSOR_DOMAIN, entry.entry_id), _new_sensor
+    )
+    async_dispatcher_connect(
+        hass,
+        SIGNAL_REMOVE_ENTITIES.format(SENSOR_DOMAIN, entry.entry_id),
+        _async_remove_old_entities,
+    )
+
+    await _async_add_ecowitt_entities(sensors)
+
+
+class EcowittSensor(EcowittEntity, SensorEntity):
+    """Base class for the ecowitt sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        device: EcoWittSensor,
+        entity_description: EcowittSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(hass, entry, device)
+        self.entity_description = entity_description
+        self.device = device
+        self._attr_native_unit_of_measurement = UOM_MAP[self.device.get_stype()]
+
+    @property
+    def native_value(self):
+        """Return the state of the sensor."""
+        if self._key not in self.data.client.last_values:
+            _LOGGER.warning(
+                "Sensor %s not in last update, check range or battery", self._key
+            )
+            return STATE_UNKNOWN
+
+        return self.device.get_value()
+
+
+class EcowittRainRateSensor(EcowittSensor):
+    """Represent an Ecowitt rain rate sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        device: EcoWittSensor,
+        entity_description: EcowittSensorEntityDescription,
+    ) -> None:
+        """Initialize ecowitt rain sensor."""
+        super().__init__(hass, entry, device, entity_description)
+        self._attr_state_class = RAIN_MAP[self.device.get_key()][0]
+        self._attr_native_unit_of_measurement = RAIN_MAP[self.device.get_key()][1]
+
+
+class EcowittLightningTimeSensor(EcowittSensor):
+    """Represent an Ecowitt lightning last strike time sensor."""
+
+    @property
+    def native_value(self):
+        """Return the state of the sensor."""
+        if self._key not in self.data.client.last_values:
+            _LOGGER.warning(
+                "Sensor %s not in last update, check range or battery", self._key
+            )
+            return STATE_UNKNOWN
+
+        # strikes are reported in UTC
+        return dt_util.as_local(
+            dt_util.utc_from_timestamp(self.device.get_value())
+        ).isoformat()
```

Comment:
Following a recent update to the core, `DEVICE_CLASS_TIMESTAMP` should now be reported as native `datetime` not as a ISO strings.